pdal/util/FileUtils.cpp

```diff
@@ -201,96 +206,108 @@ std::ostream *openExisting(const std::string& name, bool asBinary)
+    {
+        return VSI_ISDIR(sStat.st_mode);
+    }
+    else
+    {
+        return false;
+    }
 }
 
 
 bool createDirectory(const std::string& dirname)
 {
-    return fs::create_directory(toNative(dirname));
+    std::string pth = CPLCleanTrailingSlash(dirname.c_str());
+    if (VSIMkdir(pth.c_str(), 0755) != 0)
+    {
+        if (errno == EEXIST)
+            return false;
+        else
+        {
+            std::stringstream ss;
+            ss << "Unable to create directory " << dirname << std::endl << VSIStrerror(errno);
+            throw std::runtime_error(ss.str());
+        }
+    }
+    else
+        return true;
 }
 
 
 bool createDirectories(const std::string& dirname)
 {
-    // Need to strip any /'s off the end because windows and unix
-    // create_directories seem to behave differently
-    std::string s(dirname);
-    if('/' == s.back())
-        s.pop_back();
-
-    return fs::create_directories(toNative(s));
+    std::string pth = CPLCleanTrailingSlash(dirname.c_str());
+    if (VSIMkdirRecursive(pth.c_str(), 0755) != 0)
+    {
+        if (errno == EEXIST)
+            return false;
+        else
+        {
+            std::stringstream ss;
+            ss << "Unable to create directories " << dirname << std::endl << VSIStrerror(errno);
+            throw std::runtime_error(ss.str());
+        }
+    }
+    else
+        return true;
 }
 
 
 void deleteDirectory(const std::string& dirname)
 {
-    fs::remove_all(toNative(dirname));
+    VSIRmdirRecursive(dirname.c_str());
 }
 
 
 std::vector<std::string> directoryList(const std::string& dir)
 {
     std::vector<std::string> files;
-
-    try
+    CPLStringList aosFileList(VSIReadDir(dir.c_str()));
+    for (int i = 0; i < aosFileList.size(); i++)
     {
-        fs::directory_iterator it(toNative(dir));
-        fs::directory_iterator end;
-        while (it != end)
-        {
-            files.push_back(it->path().u8string());
-            it++;
-        }
-    }
-    catch (fs::filesystem_error& )
-    {
-        files.clear();
+        std::string osPath =
+            CPLFormFilename(dir.c_str(), aosFileList[i], nullptr);
+        files.push_back(osPath);
     }
+
     return files;
 }
 
 
 void closeFile(std::ostream *out)
 {
-    // An ofstream is closeable and deletable, but
-    // an ostream like &cout isn't.
-    if (!out)
-        return;
-    std::ofstream *ofs = dynamic_cast<std::ofstream *>(out);
-    if (ofs)
+    if (out)
     {
+        std::ofstream *ofs = dynamic_cast<std::ofstream *>(out);
         ofs->close();
-        delete ofs;
+        delete out;
     }
 }
 
 
 void closeFile(std::istream* in)
 {
-    // An ifstream is closeable and deletable, but
-    // an istream like &cin isn't.
-    if (!in)
-        return;
-    std::ifstream *ifs = dynamic_cast<std::ifstream *>(in);
-    if (ifs)
+    if (in)
     {
+        std::ifstream *ifs = dynamic_cast<std::ifstream *>(in);
         ifs->close();
-        delete ifs;
+        delete in;
```

Comment:
This will crash if the cast fails.

Reply:
My mistake from copying the original code. However I am going to go back to how it was, there is no need to cast and close. The file handle is closed in the unique_ptr and the stream is flushed in the destructor.